# HG changeset patch
# User Rik <rik@octave.org>
# Date 1465597626 25200
#      Fri Jun 10 15:27:06 2016 -0700
# Node ID 09583900661b5b002e93eaac5dcb63a533db4d5e
# Parent  25493c55d9761e49da8546090eaabc0a50950619
Don't draw connecting line between symbols when not specified (bug #48198).

* __gnuplot_draw_axes__.m: Check linetype variable lt and only set style to "lines"
if it is not empty.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -929,17 +929,21 @@ function __gnuplot_draw_axes__ (h, plot_
                ccol = 255*ccol*[0x1; 0x100; 0x10000];
              endif
            else
              colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
                                   round (255*color));
            endif
 
            sidx = 1;
-           style = "lines";
+           if (isempty (lt))
+             style = "";
+           else
+             style = "lines";
+           endif
            tmpwith = {};
 
            facesame = true;
            if (! isequal (pt, pt2) && isfield (obj, "markerfacecolor")
                && ! strcmp (obj.markerfacecolor, "none"))
              if (strcmp (obj.markerfacecolor, "auto")
                  || ! isnumeric (obj.markerfacecolor)
                  || (isnumeric (obj.markerfacecolor)
@@ -1834,17 +1838,21 @@ function [style, ltidx] = do_linestyle_c
   [pt, pt2, obj] = gnuplot_pointtype (obj);
 
   if (! isempty (pt))
     found_style = true;
   endif
 
   sidx = 1;
   if (isempty (errbars))
-    style{sidx} = "lines";
+    if (isempty (lt))
+      style{sidx} = "";
+    else
+      style{sidx} = "lines";
+    endif
     ltidx(sidx) = idx;
 
     facesame = true;
     if (! isequal (pt, pt2) && isfield (obj, "markerfacecolor")
         && ! strcmp (obj.markerfacecolor, "none"))
       if (strcmp (obj.markerfacecolor, "auto")
           || ! isnumeric (obj.markerfacecolor)
           || (isnumeric (obj.markerfacecolor)
